app/code/community/Razorpay/Payments/Helper/Data.php

```diff
@@ -159,6 +143,54 @@ public function sendRequest($url, $content, $method = 'POST')
         }
     }
 
+    private function getCurlHandle($url, $method = 'POST', $content = array())
+    {
+        $paymentModel = Mage::getModel(self::PAYMENT_MODEL);
+
+        $keyId = $paymentModel->getConfigData(self::KEY_ID);
+        $keySecret = $paymentModel->getConfigData(self::KEY_SECRET);
+
+        $method = strtoupper($method);
+
+        //cURL Request
+        $ch = curl_init();
+
+        curl_setopt($ch, CURLOPT_URL, $url);
+        curl_setopt($ch, CURLOPT_USERPWD, $keyId . ":" . $keySecret);
+        curl_setopt($ch, CURLOPT_TIMEOUT, self::REQUEST_TIMEOUT);
+
+        if (is_array($content))
+        {
+            $data = http_build_query($content);
+        }
+        else if (is_string($content))
+        {
+            $data = $content;
+        }
+
+        switch ($method)
+        {
+            case 'POST':
+                curl_setopt($ch, CURLOPT_POST, true);
+                curl_setopt($ch, CURLOPT_POSTFIELDS, $data);
+                break;
+            case 'PATCH':
+            case 'PUT':
+                curl_setopt($ch, CURLOPT_CUSTOMREQUEST, $method);
+                curl_setopt($ch, CURLOPT_POSTFIELDS, $data);
+                break;
+            case 'DELETE':
+                curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'DELETE');
```

Comment:
No, `CURLOPT_CUSTOMREQUEST`. Setting it to POST won't do any harm. Could be moved outside of switch.